com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ShaderCompilers.java

```diff
@@ -19,167 +19,91 @@
+            ArrayList<ShaderProgramBuilder.ShaderBuildResult> shaderBuildResults = new ArrayList<>();
+            ArrayList<ShaderDesc.Language> shaderLanguages = getPlatformShaderLanguages(shaderType, outputSpirv);
 
-                if (outputSpirv)
-                {
-                    shaderLanguages.add(ShaderDesc.Language.LANGUAGE_SPIRV);
-                }
+            assert shaderLanguages != null;
+            for (ShaderDesc.Language shaderLanguage : shaderLanguages) {
+                ShaderDesc.Shader.Builder builder = ShaderProgramBuilder.makeShaderBuilder(shaderLanguage, pipeline.crossCompile(shaderType, shaderLanguage));
+                shaderBuildResults.add(new ShaderProgramBuilder.ShaderBuildResult(builder));
             }
 
-            return getBaseShaderBuildResults(resourceOutputPath, shaderSource, shaderType, shaderLanguages.toArray(new ShaderDesc.Language[0]), "", isDebug, softFail);
+            ShaderProgramBuilder.ShaderCompileResult compileResult = new ShaderProgramBuilder.ShaderCompileResult();
+            compileResult.shaderBuildResults = shaderBuildResults;
+            compileResult.reflector = pipeline.getReflectionData();
+            return compileResult;
         }
     }
 
-    public static class WebShaderCompiler implements IShaderCompiler {
-        public ArrayList<ShaderProgramBuilder.ShaderBuildResult> compile(String shaderSource, ES2ToES3Converter.ShaderType shaderType, String resourceOutputPath, String resourceOutput, boolean isDebug, boolean outputSpirv, boolean softFail) throws IOException, CompileExceptionError {
-            ArrayList<ShaderDesc.Language> shaderLanguages = new ArrayList<ShaderDesc.Language>();
-
-            if (shaderType != ES2ToES3Converter.ShaderType.COMPUTE_SHADER) {
-                shaderLanguages.add(ShaderDesc.Language.LANGUAGE_GLES_SM300);
-                shaderLanguages.add(ShaderDesc.Language.LANGUAGE_GLES_SM100);
-            }
-
-            return getBaseShaderBuildResults(resourceOutputPath, shaderSource, shaderType, shaderLanguages.toArray(new ShaderDesc.Language[0]), "", isDebug, softFail);
-        }
-    }
-
-    public static class NXShaderCompiler implements IShaderCompiler {
-        public ArrayList<ShaderProgramBuilder.ShaderBuildResult> compile(String shaderSource, ES2ToES3Converter.ShaderType shaderType, String resourceOutputPath, String resourceOutput, boolean isDebug, boolean outputSpirv, boolean softFail) throws IOException, CompileExceptionError {
-            ArrayList<ShaderDesc.Language> shaderLanguages = new ArrayList<ShaderDesc.Language>();
-            shaderLanguages.add(ShaderDesc.Language.LANGUAGE_SPIRV);
-            return getBaseShaderBuildResults(resourceOutputPath, shaderSource, shaderType, shaderLanguages.toArray(new ShaderDesc.Language[0]), "", isDebug, softFail);
-        }
+    public static IShaderCompiler getCommonShaderCompiler(Platform platform) {
+        return new CommonShaderCompiler(platform);
     }
```

Comment:
We can potentially remove this file as well, it's a shared class now instead of separate per platform..